assets/js/base/components/price-slider/index.js

```diff
@@ -26,321 +36,269 @@ const PriceSlider = ( {
 	 */
-	const getProgressStyle = () => {
+	const getProgressStyle = useMemo( () => {
 		const low =
-			Math.round( 100 * ( ( value.min - min ) / ( max - min ) ) ) - 0.5;
+			Math.round(
+				100 *
+					( ( minPrice - minConstraint ) /
+						( maxConstraint - minConstraint ) )
+			) - 0.5;
 		const high =
-			Math.round( 100 * ( ( value.max - min ) / ( max - min ) ) ) + 0.5;
+			Math.round(
+				100 *
+					( ( maxPrice - minConstraint ) /
+						( maxConstraint - minConstraint ) )
+			) + 0.5;
 
 		return {
 			'--low': low + '%',
 			'--high': high + '%',
 		};
-	};
+	}, [ minPrice, minConstraint, maxPrice, maxConstraint ] );
+
+	/**
+	 * Trigger the onChange prop callback with new values.
+	 */
+	const triggerChange = useCallback( () => {
+		onChange( [ minPrice, maxPrice ] );
+	}, [ minPrice, maxPrice ] );
 
 	/**
 	 * Works around an IE issue where only one range selector is visible by changing the display order
 	 * based on the mouse position.
 	 *
 	 * @param {obj} event event data.
 	 */
-	const findClosestRange = ( event ) => {
-		if ( isLoading ) {
-			return;
-		}
-		const bounds = event.target.getBoundingClientRect();
-		const x = event.clientX - bounds.left;
-		const minWidth = minRange.current.offsetWidth;
-		const minValue = minRange.current.value;
-		const maxWidth = maxRange.current.offsetWidth;
-		const maxValue = maxRange.current.value;
+	const findClosestRange = useCallback(
+		( event ) => {
+			if ( isLoading ) {
+				return;
+			}
+			const bounds = event.target.getBoundingClientRect();
+			const x = event.clientX - bounds.left;
+			const minWidth = minRange.current.offsetWidth;
+			const minValue = minRange.current.value;
+			const maxWidth = maxRange.current.offsetWidth;
+			const maxValue = maxRange.current.value;
 
-		const minX = minWidth * ( minValue / max );
-		const maxX = maxWidth * ( maxValue / max );
+			const minX = minWidth * ( minValue / maxConstraint );
+			const maxX = maxWidth * ( maxValue / maxConstraint );
 
-		const minXDiff = Math.abs( x - minX );
-		const maxXDiff = Math.abs( x - maxX );
+			const minXDiff = Math.abs( x - minX );
+			const maxXDiff = Math.abs( x - maxX );
 
-		/**
-		 * The default z-index in the stylesheet as 20. 20 vs 21 is just for determining which range
-		 * slider should be at the front and has no meaning beyond
-		 */
-		if ( minXDiff > maxXDiff ) {
-			minRange.current.style.zIndex = 20;
-			maxRange.current.style.zIndex = 21;
-		} else {
-			minRange.current.style.zIndex = 21;
-			maxRange.current.style.zIndex = 20;
-		}
-	};
+			/**
+			 * The default z-index in the stylesheet as 20. 20 vs 21 is just for determining which range
+			 * slider should be at the front and has no meaning beyond
+			 */
+			if ( minXDiff > maxXDiff ) {
+				minRange.current.style.zIndex = 20;
+				maxRange.current.style.zIndex = 21;
+			} else {
+				minRange.current.style.zIndex = 21;
+				maxRange.current.style.zIndex = 20;
+			}
+		},
+		[ isLoading, maxConstraint, minRange, maxRange ]
```

Comment:
`minRange` and `maxRange` don't need to be dependencies here because they are refs (so you always have access to the current value).